### PR TITLE
Add documentation when running together with other LoadBalancer controllers

### LINKED ISSUE #104: [Proposal] Promote and/or document the use of the Tunnel CRD
Some people reported to have issues when running the inlets-operator in combination with another LoadBalancer controller like MetalB. When having multiple LoadBalancer present at the same time, whenever a Service type LoadBalancer is created, BOTH LBs will provide an address.

Although the inlets-operator has a flag `-annotated-only`, must other LoadBalancer controller do not have such kind of feature and will compete with the inlets-operator.

The inlets-operator can also create the required exit-node and client deployment without the need of a LoadBalancer service, but by creating a Tunnel resource. This is not really documented and can be a proper solution when running together with other LoadBalancer controllers.

E.g:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 2
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
        - name: nginx
          image: nginx
          ports:
            - containerPort: 80
              protocol: TCP
---
apiVersion: v1
kind: Service
metadata:
  name: nginx
spec:
  type: ClusterIP
  ports:
  - name: http
    port: 80
    targetPort: 80
  selector:
    app: nginx
---
apiVersion: inlets.inlets.dev/v1alpha1
kind: Tunnel
metadata:
  name: nginx
spec:
  serviceName: nginx
  auth_token: <token>
```

## Expected Behaviour

The inlets-operator can be deployed together with other LoadBalancer controllers without having them compete against each other.

## Current Behaviour

When having multiple LoadBalancer present, both will provide an address for each Service of type LoadBalancer

## Possible Solution

Promote and document the usage of Tunnel resource in combination with the `annotated-only` flag
Optional: change the current behaviour by making the `-annotated-flag` to `true` by default

## Steps to Reproduce (for bugs)

1. install inlets-operator and MetalB
2. create a service of type LoadBalancer
3. both controllers will act on the created service

## Context

Related issues:

https://github.com/inlets/inlets-operator/issues/95
https://github.com/kubernetes/kubernetes/issues/91306
https://github.com/metallb/metallb/issues/685

## Your Environment

* inlets-operator version, find via `kubectl get deploy inlets-operator -o wide`

* Kubernetes distribution i.e. minikube v0.29.0., KinD v0.5.1, Docker Desktop:

* Kubernetes version `kubectl version`:

* Operating System and version (e.g. Linux, Windows, MacOS):

* Cloud provisioner: (DigitalOcean.com / Packet.com / etc)

### PR DESCRIPTION
**Pull request by jsiebens**
- [x] I have raised an issue to propose this change.
Fixes #104

## Description

Add a small section in the README on how to run together with another LoadBalancer controller like MetalLB

## How Has This Been Tested?

Installed the inlets operator and exposed a service by using a Tunnel resource instead of a LoadBalancer service.

```
apiVersion: v1
kind: Service
metadata:
  name: nginx
spec:
  type: ClusterIP
  ports:
  - name: http
    port: 80
    targetPort: 80
  selector:
    app: nginx
---
apiVersion: inlets.inlets.dev/v1alpha1
kind: Tunnel
metadata:
  name: nginx
spec:
  serviceName: nginx
  auth_token: <token>
```

## How are existing users impacted? What migration steps/scripts do we need?

No impact

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +166,40 @@ There are two ways to override the behaviour:
 
   You can also set the operator to ignore the services by default and only manage them when the annotation is true. `dev.inlets.manage=true` with the flag `-annotated-only`
 
+## Running in combination with other LoadBalancer controllers
+
+By default the operator will create a tunnel for every LoadBalancer service, which can cause issues when running in combination with other LoadBalancer controllers like MetalLB.
+
+When running the inlets-operator with the `-annotated-only` flag, the inlets-operator will ignore all LoadBalancer services, unless explicitly annotated with `dev.inlets.manage=true`, and will not compete against the other LoadBalancer controllers.
+
+The operator is still able to expose services in this mode, by using a ClusterIP service with a Tunnel resource.
```
Comment (alexellis) [inline README.md:175]: In this example, how do you find out the server IP?

Comment (jsiebens) [reply README.md:175]: funny enough, the ClusterIP service is actually updated with the server IP by the inlets-operator

Comment (alexellis) [reply README.md:175]: Can you explain that and show an example command from kubectl?

Comment (alexellis) [reply README.md:175]: This section may be better under `Annotations and ignoring services`

Or maybe as the primary title for `Annotations and ignoring services` and then we have three sub-titles?

**Review by alexellis — APPROVED** [review]
Approved. LGTM

**Pull request merged**